contentcuration/contentcuration/frontend/channelEdit/views/ImportFromChannels/SearchOrBrowseWindow.vue

```diff
@@ -564,6 +707,125 @@
           ...this.otherRecommendations.slice(0, limit),
         ];
       },
+      initialRecommendationsEvent(request, response) {
+        const nodeToRank = {};
+        const recommendedNodes = response || [];
+        recommendedNodes.forEach(node => {
+          nodeToRank[node.node_id] = node.rank;
+        });
+
+        const content = [...this.recommendations, ...this.otherRecommendations];
+        this.recommendationsEvent = new RecommendationsEvent({
+          context: { request },
+          contentnode_id: this.importDestinationFolder.id,
+          content_id: this.importDestinationFolder.content_id,
+          target_channel_id: this.importDestinationFolder.channel_id,
+          user_id: this.userId,
+          content: content.map(node => ({
+            content_id: node.content_id,
+            node_id: node.id,
+            channel_id: node.channel_id,
+            rank: nodeToRank[node.id],
+          })),
+        });
+      },
+      handleNotRelevantRecommendation(node) {
+        const interactionEvent = new RecommendationsInteractionEvent({
+          recommendation_event_id: this.recommendationsEvent.id,
+          contentnode_id: node.id,
+          content_id: node.content_id,
+          context: {
+            //ToDo: Add appropriate context to be sent with the interaction event
+          },
```

Comment:
I believe for the `context`, we want what is passed to `fetchRecommendations`:
```js
const data = {
  ...this.embedTopicRequest,
  override_threshold: belowThreshold,
};
```

Is that correct @ozer550 ?

Reply:
We didn't decide anything particularly for the context field of Interaction event! Having this makes sense but, I'm not sure if that would have a meaning storing particularly in Interation event object? This may seems more appropriate to store in the context of RecommenedationsEvent instead?

Reply:
This is what had something like in my mind for Interaciton event object:  _**this can be a flexible JSON blob with any additional context about the event (e.g. if we allowed them to make notes about why it wasn’t relevant, that could go here – or for a “show_more” action, how many more items were shown)**_

Reply:
@ozer550, Okay, I see. But aren't we capturing the "why" with the `feedback_reason` property? Would that mean that we don't need addition context to be stored for this event?

Reply:
The differentiation that I had in my mind was that we can store the specific written feedback given by users in context, and then the pre-existing radio button feedback would go in feedback_reason field.